File: weddy-backend-1.0/weddy/src/main/java/com/project/weddy/domain/user/dto/response/AuthResponse.java
package com.project.weddy.domain.user.dto.response;

import lombok.Builder;
import lombok.Getter;

/**
 * 인증 응답 DTO.
 * 로그인, 회원가입, 토큰 갱신 후 클라이언트에 전달되는 데이터를 담는다.
 */
@Getter
@Builder
public class AuthResponse {

    /** JWT 액세스 토큰 */
    private String accessToken;

    /** JWT 리프레시 토큰 */
    private String refreshToken;

    /** 사용자 엔티티 PK (14자리 숫자 문자열) */
    private String userOid;

    /** 로그인 아이디 */
    private String userId;

    /** 사용자 이름 */
    private String name;

    /** 사용자 역할 (GROOM / BRIDE) */
    private String role;
}

File: weddy-frontend-1.0/weddy/lib/features/auth/data/model/auth_response.dart
/// 서버 /auth/signup, /auth/login, /auth/refresh 공통 응답 DTO.
///
/// ApiResponse<AuthResponse> 의 data 필드로 전달된다.
class AuthResponse {
  final String accessToken;
  final String refreshToken;
  final String userOid;
  final String userId;
  final String name;
  final String role;

  const AuthResponse({
    required this.accessToken,
    required this.refreshToken,
    required this.userOid,
    required this.userId,
    required this.name,
    required this.role,
  });

  factory AuthResponse.fromJson(Map<String, dynamic> json) {
    return AuthResponse(
      accessToken: json['accessToken'] as String,
      refreshToken: json['refreshToken'] as String,
      userOid: json['userOid'] as String,
      userId: json['userId'] as String,
      name: json['name'] as String,
      role: json['role'] as String,
    );
  }

  @override
  String toString() {
    return 'AuthResponse('
        'userOid: $userOid, '
        'userId: $userId, '
        'name: $name, '
        'role: $role'
        ')';
  }
}
